clients/cloud/rust/src/producer.rs

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+extern crate futures;
+extern crate rdkafka;
+
+use futures::future::Future;
+use rdkafka::error::KafkaError;
+use rdkafka::message::OwnedMessage;
+use rdkafka::producer::{FutureProducer, FutureRecord};
+use std::boxed::Box;
+
+mod utils;
+
+fn log_produce_result(
+    topic: String,
```

Comment:
Does topic here need to be owned or can it be `&str`?

Reply:
You are right, it doesn't need to be owned!